pkg/rpc/dial.go

```diff
@@ -111,6 +114,54 @@ func (d Dialer) DialAddressID(ctx context.Context, address string, id storj.Node
 	return d.dial(ctx, address, d.TLSOptions.ClientTLSConfig(id))
 }
 
+// DialAddressInsecureBestEffort is like DialAddressInsecure but tries to dial a node securely if
+// it can.
+//
+// nodeURL is like a storj.NodeURL but (a) requires an address and (b) does not require a
```

Comment:
`storj.NodeURL` doesn't have any restriction on the address nor URL. You could still use `storj.NodeURL` here and then check whether `if nodeurl.Address == "" { return Error.New }`

Reply:
i believe it uses a storj.NodeID, which i would hope has a length check. this has no length check, so i didn't want to use a storj.NodeID for a node id prefix

Reply:
ah, missed the prefix part